src/main/java/org/tron/p2p/connection/socket/PeerClient.java

```diff
@@ -41,7 +41,9 @@ public void close() {
   public void connect(String host, int port, String remoteId) {
     try {
       ChannelFuture f = connectAsync(host, port, remoteId, false, false);
```

Comment:
Could null be returned here?

Reply:
Yes, when shutdown is hook,  this will return null

Reply:
There is no important data processing, so why add a shutdown hook?

Reply:
where is the shutdown hook?